src/poprox_recommender/components/diversifiers/locality_calibration.py

```diff
@@ -1,19 +1,38 @@
+import datetime
+from collections import defaultdict
+
+import numpy as np
 import torch as th
+import time
+from openai import OpenAI
+from sentence_transformers import SentenceTransformer
+from sklearn.metrics.pairwise import cosine_similarity
+
+from poprox_concepts import Article, ArticleSet, InterestProfile
+from poprox_recommender.components.diversifiers.calibration import compute_kl_divergence
+from poprox_recommender.lkpipeline import Component
+from poprox_recommender.topics import extract_general_topics, extract_locality, normalized_category_count
 
-from poprox_concepts import ArticleSet, InterestProfile
-from poprox_recommender.components.diversifiers.calibration import Calibrator
-from poprox_recommender.topics import extract_locality, normalized_category_count
+# Only uncomment this in offline theta value exploration
+# KL_VALUE_PATH = '/home/sun00587/research/News_Locality_Polarization/poprox-recommender-locality/outputs/theta_kl_values_11-17.txt'
 
+MAX_RETRIES = 3
+DELAY = 2
+SEMANTIC_THRESHOLD = 0.2
 
-# Locality Calibration uses MMR
-# to rerank recommendations according to
-# locality calibration
-class LocalityCalibrator(Calibrator):
-    def __init__(self, theta: float = 0.1, num_slots=10):
-        super().__init__(theta, num_slots)
+class LocalityCalibrator(Component):
+    def __init__(self, theta_local: float = 0.1, theta_topic: float = 0.1, num_slots=10):
```

Comment:
I don't think we're using these class level variables anymore for the theta values